core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -150,10 +150,10 @@ class Partition(val topic: String,
     * @return true iff the future replica is created
     */
   def maybeCreateFutureReplica(logDir: String): Boolean = {
-    // The readLock is needed to make sure that while the caller checks the log directory of the
+    // The writeLock is needed to make sure that while the caller checks the log directory of the
     // current replica and the existence of the future replica, no other thread can update the log directory of the
     // current replica or remove the future replica.
-    inReadLock(leaderIsrUpdateLock) {
+    inWriteLock(leaderIsrUpdateLock) {
```

Comment:
Hmm, could you explain why we need a write lock here? This method doesn't change any state value. So, intuitive, it only needs a read lock.

Reply:
Yeah the current implementation does not require write lock. But it seems intuitive to use the write lock because changes the state here -- a future replica will be created for this partition. The code might be more consistent by using a write lock so that we always use a write lock whenever we add or remove a replica.